### PR TITLE
Fixed up GaugeReward test

### PR DESCRIPTION
**Pull request by asselstine**

### CODE REVIEW

File: contracts/GaugeReward.sol [hunk 1]
```diff
@@ -123,12 +129,20 @@ contract GaugeReward is IGaugeReward {
...
-    constructor(IGaugeController _gaugeController, IVault _vault) {
+    constructor(
+        IGaugeController _gaugeController,
+        address _vault,
+        address _liquidator,
+        uint32 _stakerCut
+    ) {
         require(address(_gaugeController) != address(0), "GReward/GC-not-zero-address");
         require(address(_vault) != address(0), "GReward/Vault-not-zero-address");
+        require(_stakerCut < 1e9, "GReward/staker-cut-lt-1e9");
```
Comment (PierrickGT) [inline contracts/GaugeReward.sol:140]: Add a require for `_liquidator` different than address zero.

File: contracts/GaugeReward.sol [hunk 2]
```diff
@@ -123,12 +129,20 @@ contract GaugeReward is IGaugeReward {
...
 
         gaugeController = _gaugeController;
         vault = _vault;
+        stakerCut = _stakerCut;
+        liquidator = _liquidator;
 
         emit Deployed(_gaugeController, _vault);
```
Comment (PierrickGT) [inline contracts/GaugeReward.sol:147]: Emit `_liquidator` and `_stakerCut` in event.

File: contracts/GaugeReward.sol [hunk 3]
```diff
@@ -148,46 +162,40 @@ contract GaugeReward is IGaugeReward {
...
-    ) external {
-        address _vaultAddress = address(vault);
-
-        if (_token != _currentRewardToken(_gauge).token) {
-            vault.increaseERC20Allowance(
-                _token,
-                address(this),
-                type(uint256).max - _token.allowance(_vaultAddress, address(this))
-            );
+    function afterSwap(IPrizePool prizePool, ITicket ticket, uint256 ticketAmount, IERC20 token, uint256 tokenAmount) external override {
```
Comment (PierrickGT) [inline contracts/GaugeReward.sol:171]: Remove `prizePool` and `ticketAmount` params if not used.

File: contracts/GaugeReward.sol [hunk 4]
```diff
@@ -196,10 +204,11 @@ contract GaugeReward is IGaugeReward {
         address _gauge,
         address _user,
         uint256 _oldStakeBalance
-    ) external override {
+    ) external override onlyGaugeController {
         RewardToken memory _rewardToken = _claimPastRewards(_gauge, _user, _oldStakeBalance);
-
-        _claim(_gauge, _rewardToken.token, _user, _oldStakeBalance, false);
+        if (_rewardToken.token != IERC20(address(0))) {
```
Comment (PierrickGT) [inline contracts/GaugeReward.sol:209]:
```suggestion
        if (address(_rewardToken.token) != address(0)) {
```

File: contracts/GaugeReward.sol [hunk 5]
```diff
@@ -300,20 +315,22 @@ contract GaugeReward is IGaugeReward {
...
+                i = i - 1;
                 _rewardToken = gaugeRewardTokens[_gauge][i];
 
-                if (i == _gaugeRewardTokenslength - 1) {
+                if (i == _gaugeRewardTokensLength - 1) {
                     _latestRewardToken = _rewardToken;
                 }
 
-                if (_rewardToken.timestamp > _userLastClaimedTimestamp) {
+                if (_rewardToken.timestamp > _userLastClaimedTimestamp && _userLastClaimedTimestamp > 0) {
```
Comment (PierrickGT) [inline contracts/GaugeReward.sol:333]: Check for `0` first.
```suggestion
                if (_userLastClaimedTimestamp > 0 && _rewardToken.timestamp > _userLastClaimedTimestamp) {
```

**Review by PierrickGT — CHANGES_REQUESTED** [review]

File: contracts/GaugeReward.sol [hunk 6]
```diff
@@ -148,46 +162,40 @@ contract GaugeReward is IGaugeReward {
...
-        address _vaultAddress = address(vault);
-
-        if (_token != _currentRewardToken(_gauge).token) {
-            vault.increaseERC20Allowance(
-                _token,
-                address(this),
-                type(uint256).max - _token.allowance(_vaultAddress, address(this))
-            );
+    function afterSwap(IPrizePool prizePool, ITicket ticket, uint256 ticketAmount, IERC20 token, uint256 tokenAmount) external override {
+        require(msg.sender == liquidator, "GReward/only-liquidator");
```
[the remark this answers is not in the document]
Comment (asselstine) [reply contracts/GaugeReward.sol:172]: That code isn't used anywhere else- refactoring it into a modifier doesn't make sense

File: contracts/GaugeReward.sol (hunk 4 above)
Comment (asselstine) [reply contracts/GaugeReward.sol:209]: I mean....same difference right?

**Review by PierrickGT — APPROVED** [review]

**Pull request merged**